### PR TITLE
Bug fix/make aql fast again 8

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *all AQL tests*.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10578/

### CODE REVIEW

File: arangod/Aql/DependencyProxy.cpp [hunk 1]
```diff
@@ -166,9 +166,7 @@ DependencyProxy<blockPassthrough>::fetchBlock(size_t atMost) {
 
   TRI_ASSERT(!_blockQueue.empty());
 
-  ExecutionState state;
-  SharedAqlItemBlockPtr block;
-  std::tie(state, block) = _blockQueue.front();
+  auto&& [state, block] = _blockQueue.front();
   _blockQueue.pop_front();
```
Comment (goedderz) [inline arangod/Aql/DependencyProxy.cpp:170]: This looks invalid. You can maybe do something like
```
auto [state, block] = std::move( _blockQueue.front());
```
though.

File: arangod/Aql/DependencyProxy.cpp [hunk 2]
```diff
@@ -166,9 +166,7 @@ DependencyProxy<blockPassthrough>::fetchBlock(size_t atMost) {
 
   TRI_ASSERT(!_blockQueue.empty());
 
-  ExecutionState state;
-  SharedAqlItemBlockPtr block;
-  std::tie(state, block) = _blockQueue.front();
+  auto&& [state, block] = _blockQueue.front();
```
Comment (jsteemann) [inline arangod/Aql/DependencyProxy.cpp:169]:
```suggestion
  auto [state, block] = std::move(_blockQueue.front());
```

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10597/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10601/

**Comment by jsteemann** [comment]
Tests blue

File: arangod/Aql/DependencyProxy.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/Aql/DependencyProxy.cpp]: Ok, I adjustedt this.

File: arangod/Aql/MultiAqlItemBlockInputRange.cpp [hunk 3]
```diff
@@ -35,12 +35,12 @@ using namespace arangodb::aql;
...
 namespace {
   static auto RowHasNonEmptyValue(ShadowAqlItemRow const& row) -> bool {
-    if (!row.isInitialized()) {
-      return false;
-    }
-    for (RegisterId registerId = 0; registerId < row.getNrRegisters(); ++registerId) {
-      if (!row.getValue(registerId).isEmpty()) {
-        return true;
+    if (row.isInitialized()) {
+      RegisterId const numRegisters = static_cast<RegisterId>(row.getNrRegisters());
```
Comment (goedderz) [inline arangod/Aql/MultiAqlItemBlockInputRange.cpp:39]:
```suggestion
      auto const numRegisters = row.getNrRegisters();
```
should also work, if I'm not mistaken. But it's not important.

**Pull request merged**